src/Workspaces/CSharp/Portable/Indentation/CSharpIndentationService.cs

```diff
@@ -47,12 +43,13 @@ protected override AbstractFormattingRule GetSpecializedIndentationFormattingRul
             CompilationUnitSyntax root,
             TextLine line,
             OptionSet optionSet,
-            CancellationToken cancellationToken)
+            out SyntaxToken token)
```

Comment:
hmm... what is this token for?

Reply:
if we decide that we want to use the SmartTokenFormatter, then this passes out the actual token for the smar ttoken formatter to actually format:

```c#
            public bool TryGetSmartTokenIndentation(out IndentationResult indentationResult)
            {
                indentationResult = default;
                if (!ShouldUseTokenIndenter(out var token))
                {
                    return false;
                }

                // var root = document.GetSyntaxRootSynchronously(cancellationToken);
                var sourceText = Tree.GetText(CancellationToken);

                var formatter = CreateSmartTokenFormatter();
                var changes = formatter.FormatTokenAsync(Document.Project.Solution.Workspace, token, CancellationToken)
                                       .WaitAndGetResult(CancellationToken);
```

Reply:
I've doc'ed the ShouldUseTokenIndenter function to make this contract documented.